kubernetes/kubernetes.go

```diff
@@ -186,14 +185,17 @@ func (ka KubeAgentConfig) collectMetrics(
 	// get raw Heapster metric sample
 	hme, err := rawClient.GetRawEndPoint("heapster-metrics-export", metricSampleDir, config.HeapsterURL)
 	if err != nil {
-		return fmt.Errorf("error retrieving Raw Heapster metrics: %s", err)
+		return fmt.Errorf("unable to retrieve raw heapster metrics: %s", err)
 	}
 
 	defer util.SafeClose(hme.Close, &rerr)
 
-	err = handleBaselineHeapsterMetrics(msd, baselineMetricSample, hme.Name())
-	if err != nil {
-		return err
+	baselineMetricSample, err := util.MatchOneFile(path.Dir(config.msExportDirectory.Name()), "/baseline-metrics-export*")
+	if err == nil {
```

Comment:
I would refactor this to instead be:

`if err != nil {
  return err
}
err = handleBaselineHeapsterMetrics(msd, baselineMetricSample, hme.Name())
if err != nil {
  return err
}
`

Reply:
unless you want to continue through the rest of the function if it fails.

Reply:
We want it to copy the baseline if one is found else continue through.